src/main/scala/viper/silver/plugin/standard/inline/InlinePredicatePlugin.scala

```diff
@@ -0,0 +1,42 @@
+package viper.silver.plugin.standard.inline
+
+import viper.silver.ast.{Exp, LocalVar, Method, Node, PredicateAccess, Program}
+import viper.silver.plugin.{ParserPluginTemplate, SilverPlugin}
+
+class InlinePredicatePlugin extends SilverPlugin with ParserPluginTemplate {
+
+  private[this] val InlinePredicateKeyword = "inline"
+
+  override def beforeVerify(input: Program): Program = {
+    val declaredMethods = input.methods
+    val methodsWithExpandedPres = declaredMethods.map(expandPreconditions(_, input))
+
+    // TODO: actually rewrite program with methods with expanded predicates
+    input
+  }
+
+  private[this] def expandPreconditions(method: Method, program: Program): Method = {
+    method.copy(name = method.name,
+      formalArgs = method.formalArgs,
+      formalReturns = method.formalReturns,
+      pres = method.pres.map { pre =>
```

Comment:
We take each `pres` (which I'm assuming to be a precondition predicate), and we map it to the actual body (only if we can expand it successfully)